### PR TITLE
Integrate iOS SDK

### PR DESCRIPTION
**Pull request by gaguirre**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -20,9 +23,6 @@ repositories {
     flatDir {
         dirs 'libs'
     }
-    maven {
-        url "$rootDir/../../_packages/node_modules/react-native/android"
-    }
```
Comment (aschenkel) [inline android/build.gradle]: we could avoid commiting this file

File: android/src/main/java/io/underscope/RNSwrveModule.java [hunk 2]
```diff
@@ -121,6 +124,7 @@ public void userUpdate(ReadableMap attributes) {
     SwrveManager.getInstance().logIAPAction(analytics);
   }
 
+  @Deprecated
```
Comment (aschenkel) [inline android/src/main/java/io/underscope/RNSwrveModule.java]: so we **shouldnt** use `iap`?

File: android/src/main/java/io/underscope/RNSwrveModule.java [hunk 3]
```diff
@@ -138,12 +142,53 @@ public void logFailedReceiptAction(String productId, String error) {
...
+    String subtype = null;
+    String subtype2 = null;
+
+    if (options != null) {
+      level = options.hasKey("level") ? options.getInt("level") : null;
+      context = options.hasKey("context") ? options.getString("context") : null;
+      type = options.hasKey("type") ? options.getString("type") : null;
+      subtype = options.hasKey("subtype") ? options.getString("subtype") : null;
+      subtype2 = options.hasKey("subtype2") ? options.getString("subtype2") : null;
+    }
```
Comment (aschenkel) [inline android/src/main/java/io/underscope/RNSwrveModule.java]: could we do?

```java

if (options == null) {
   options = new ReadableMap()
}

Integer level = options.hasKey("level") ? options.getInt("level") : null;
String context = options.hasKey("context") ? options.getString("context") : null;
String type = options.hasKey("type") ? options.getString("type") : null;
String subtype = options.hasKey("subtype") ? options.getString("subtype") : null;
String subtype2 = options.hasKey("subtype2") ? options.getString("subtype2") : null;
  
```

maybe we cant

File: ios/RNSwrve.m [hunk 4]
```diff
@@ -25,7 +27,7 @@ - (dispatch_queue_t)methodQueue
 
 RCT_EXPORT_METHOD(sessionEnd)
 {
-    RCTLogInfo(@"Called sessionEnd");
+    // TODO: track type `session_end`
```
Comment (aschenkel) [inline ios/RNSwrve.m]: is it okey to leave this as a TODO?
just wondering

File: android/build.gradle (hunk 1 above)
Comment (ignacioola) [reply android/build.gradle]: Agree, we should see if we could take this from the `local.properties` file and include it only if we are in development.

File: android/src/main/java/io/underscope/RNSwrveModule.java (hunk 2 above)
Comment (ignacioola) [reply android/src/main/java/io/underscope/RNSwrveModule.java]: No, `iap` is called from `logIAPAction`

**Pull request merged**